### PR TITLE
Fix MEMPOOL_MARKER and MEMPOOL_SANITY_CHECK macros

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

**Comment by jforissier** [comment]
This looks fine, but libmpa does not seem to build with `-DDEBUG=1` so what's the point?
```
$ make out/arm-plat-vexpress/user_ta-lib/libmpa/libmpa.a CFLAGS="-DDEBUG=1" 
  CHK     out/arm-plat-vexpress/core/conf.mk
  UPD     out/arm-plat-vexpress/core/conf.mk
  CHK     out/arm-plat-vexpress/core/include/generated/conf.h
  UPD     out/arm-plat-vexpress/core/include/generated/conf.h
  CC      out/arm-plat-vexpress/user_ta-lib/libmpa/mpa_misc.o
In file included from lib/libmpa/mpa_misc.c:34:0:
lib/libmpa/include/mpa_debug.h:68:2: error: expected identifier or ‘(’ before ‘}’ token
  } while (0)
  ^
lib/libmpa/include/mpa_debug.h:68:4: error: expected identifier or ‘(’ before ‘while’
  } while (0)
    ^
lib/libmpa/include/mpa_debug.h:74:4: error: expected identifier or ‘(’ before ‘while’
  } while (0)
    ^
mk/compile.mk:141: recipe for target 'out/arm-plat-vexpress/user_ta-lib/libmpa/mpa_misc.o' failed
make: *** [out/arm-plat-vexpress/user_ta-lib/libmpa/mpa_misc.o] Error 1
```

Moreover, libtomcrypt also has compile errors when DEBUG is defined.
Should this PR be "Fix compile with -DDEBUG=1" instead?

**Comment by pascal-brand-st-dev** [comment]
Updated

File: lib/libmpa/include/mpa_debug.h [hunk 1]
```diff
@@ -58,19 +58,19 @@ void __mpa_dbg_dump_mpanum(const mpanum val);
 
 #define DPRINT   \
 	DPRINT_WHERE \
-	__mpa_dbg_print
+	DMSG_RAW
```
Comment (jforissier) [inline lib/libmpa/include/mpa_debug.h]: This is inconsistent. `DPRINT_WHERE` calls `__mpa_dbg_print_header()` which does `MSG_RAW(...)`. As a result, the file/line information is printed whenever `TRACE_LEVEL > 0` but the rest of the message is shown only when `TRACE_LEVEL >= TRACE_DEBUG`.
I see two options: either keep `__mpa_dbg_print()`, or replace `MSG_RAW()` with `DMSG_RAW()` in all `__mpa_dbg_{print,dump}*()` functions.

Comment (pascal-brand-st-dev) [inline lib/libmpa/include/mpa_debug.h]: What about replacing this `DMSG_RAW` into `MSG_RAW`?

**Comment by jforissier** [comment]
I don't really see the point of this PR because the debug features of libmpa are still not well integrated with the "regular" OP-TEE debug mechanism. That is, MPA debug traces are not turned on when `CFG_TEE_CORE_LOG_LEVEL` or `CFG_TEE_TA_LOG_LEVEL` is set to a value higher or equal to `TRACE_DEBUG`. What is also confusing is that the `DEBUG` macro in libmpa is not enabled automatically when `DEBUG=1` in `mk/config.mk`.

What if we would simply drop all the libmpa-specific debug macros/functions and use `DMSG` etc. instead?

**Comment by pascal-brand-st-dev** [comment]
As discussed with @jforissier , the best is to remove all the traces (which is legacy code) in libmpa as they are not used right now, and do not think that useful in current state. Will open a new pull-request with this.

**Pull request closed without merging**